llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -525,7 +524,7 @@ RegInterval WaitcntBrackets::getRegInterval(const MachineInstr *MI,
   else
     return {-1, -1};
 
-  const TargetRegisterClass *RC = TII->getOpRegClass(*MI, OpNo);
+  const TargetRegisterClass *RC = TRI->getPhysRegBaseClass(Op.getReg());
```

Comment:
We should probably be asserting this is a physical register here?
i.e. `assert(!Op.getReg().isVirtual())`

`getPhysRegBaseClass` will return null for virtuals, etc, so alternatively/additionally assert RC is valid?

Reply:
In practice I think this would have failed earlier, at the call to `TRI->getEncodingValue` above. Do you still think an explicit assert is needed somewhere?

Reply:
I think it's fine without an assert, it's not like any behaviour was changed, and misuse is unlikely.

I am only raising it because we can clearly can identify non physicals here based on `Register` type, where as most of the TRI methods are working on `MCRegister` and only identify out-of-bound registers.
We can imagine `Register` to `MCRegister` truncation yielding valid hardware registers for virtuals.